### PR TITLE
Checked if quote object contains id before looking for quote items

### PR DESCRIPTION
**Pull request by rav-redchamps**
### Description (*)

Checked if quote object contains id before looking for quote items so that it doesn’t pull all the quote items
### Fixed Issues (if relevant)

1. magento/magento2#26437: Viewing customer shopping cart in admin shows all products in catalog when there is no active quote

### Manual testing scenarios (*)

1. Create a new customer account in storefront
2. Login to admin panel and go to Customers > All Customers path
3. Edit newly created customer and go to "Shopping Cart" tab
4. You will observe that it lists many items but the shopping cart of the customer in the storefront is blank
5. But if you don't see all items here, go to database table "quote" & search it for customer id, if any quote found delete that
6. Now, refresh the customer edit page in backend & go to "Shopping Cart" tab again

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rav-redchamps** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @rav-redchamps. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @rav-redchamps, here is your [Magento instance](https://i-26489-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-26489-2-4-develop.instances.magento-community.engineering/admin_4b8d](https://i-26489-2-4-develop.instances.magento-community.engineering/admin_4b8d)
Login: `806e4ca0` Password: `f2e5a4168c5b`
Instance will be terminated in up to 3 hours.

**Comment by novikor** [comment]
@magento run Functional Tests EE

**Review by novikor — APPROVED** [review]
Reviewed and tested.
Thank you for your contribution!

**Comment by magento-engcom-team** [comment]
Hi @novikor, thank you for the review.
[ENGCOM-6667](https://jira.corp.magento.com/browse/ENGCOM-6667) has been created to process this Pull Request
:eight_spoked_asterisk: @novikor, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed

**Comment by novikor** [comment]
@magento run Functional Tests B2B

**Review by VladimirZaets — DISMISSED** [review]
@novikor Hi, thanks for the review. Why do you think that this code not required to be covered? as for me, we should cover this case

**Comment by novikor** [comment]
@VladimirZaets - I agree. However, it was not easy to decide: on one hand, this functionality is not covered at all, on another - I have doubts about whether the contributor will provide functional tests.

**Comment by novikor** [comment]
Hi, @rav-redchamps.
Would you like to implement test coverage for your PR?

**Comment by rav-redchamps** [comment]
> Hi, @rav-redchamps.
> Would you like to implement test coverage for your PR?

Hi @novikor,

Not sure, if I can spare time for this in the near future. If test coverage can be implemented later  & the fix can be merged as of now, that would be great.

**Comment by novikor** [comment]
@rav-redchamps - Okay, thank you for your contribution. 
The PR will be suspended until the tests are provided.
Unfortunately, there is no possibility to approve the PR without test coverage according to the Magento Definition Of Done.

I will not close the PR in the nearest future and hope it will be merged one day.

**Comment by novikor** [comment]
Integration or functional tests are applicable to this PR.

**Comment by rav-redchamps** [comment]
Okay, I will put the fix in the form of an opensource extension under a separate GitHub repository so that people can use it in the meantime.

**Comment by novikor** [comment]
@rav-redchamps, I am not sure what you mean.
Only Magento tests which take part in test builds apply: 
image

See https://devdocs.magento.com/guides/v2.3/test/testing.html.
Among them are Functional, Api functional, Integration, Unit.

I recommend to take a look at integration ones: https://devdocs.magento.com/guides/v2.3/test/integration/integration_test_execution.html

**Comment by rav-redchamps** [comment]
@novikor I mean this https://github.com/redchamps/customer-shopping-carts-tab-fix

**Comment by novikor** [comment]
@rav-redchamps - ok, fine then. In case all the changes will be present in current PR as well)
Thank you.

**Comment by engcom-Golf** [comment]
HI @rav-redchamps i will take care of test coverage.

**Comment by rav-redchamps** [comment]
@engcom-Golf Awesome, Thanks!

**Comment by engcom-Golf** [comment]
HI @novikor could you please review test coverage ?

File: dev/tests/integration/testsuite/Magento/Customer/Block/Adminhtml/Edit/Tab/CartTest.php [hunk 1]
```diff
@@ -5,32 +5,52 @@
...
 
 /**
  * Magento\Customer\Block\Adminhtml\Edit\Tab\Cart
  *
  * @magentoAppArea adminhtml
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  */
 class CartTest extends \PHPUnit\Framework\TestCase
 {
     const CUSTOMER_ID_VALUE = 1234;
```
Comment (novikor) [inline dev/tests/integration/testsuite/Magento/Customer/Block/Adminhtml/Edit/Tab/CartTest.php:24]: Seems to be OK.
@engcom-Golf , - thank you.

**Review by novikor — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @novikor, thank you for the review.
[ENGCOM-6667](https://jira.corp.magento.com/browse/ENGCOM-6667) has been created to process this Pull Request

**Comment by engcom-Golf** [comment]
Failure of  functional test, not related to the changes in this PR , test failing on 2.4-develop branch
DeepinScreenshot_select-area_20200131114202

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Pull request merged**